app/controllers/auth/AuthAction.scala

```diff
@@ -47,56 +47,23 @@ class AuthActionImpl @Inject()(override val authConnector: DefaultAuthConnector,
         HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
 
       authorised(ConfidenceLevel.L50 and (Enrolment("IR-SA") or Enrolment("IR-SA-AGENT")))
-        .retrieve(Retrievals.allEnrolments and Retrievals.externalId and Retrievals.credentials) {
-          case Enrolments(enrolments) ~ Some(externalId) ~ Some(credentials) => {
+        .retrieve(
+          Retrievals.allEnrolments and Retrievals.externalId and Retrievals.credentials and Retrievals.saUtr and Retrievals.confidenceLevel) {
```

Comment:
Do we need the confidenceLevel in this auth request? I thought it was only being used in the merge page?

Reply:
we need to retrieve it to be able to display the ivUplift link conditionally when a user has got paye data but is CL less than 200

Reply:
I think it's been put in here because the original SA specific `AuthenticatedRequest` has been reused, I think it can be removed from here if the MergePageAuthAction had it's own "special" auth request, this is a minor thing btw if it is too much work just leave it